datahub-web-react/src/app/entity/shared/entityForm/prompts/StructuredPropertyPrompt/UrnInput/SearchSelectUrnInput.tsx

```diff
@@ -0,0 +1,161 @@
+import { Icon } from '@src/alchemy-components';
+import { useHydratedEntityMap } from '@src/app/entityV2/shared/tabs/Properties/useHydratedEntityMap';
+import { EntityLink } from '@src/app/homeV2/reference/sections/EntityLink';
+import { Skeleton } from 'antd';
+import React, { useEffect, useMemo, useState } from 'react';
+import styled from 'styled-components';
+import { EntityType } from '../../../../../../../types.generated';
+import { SearchSelect } from '../../../../../../entityV2/shared/components/styled/search/SearchSelect';
+import { EntityAndType } from '../../../../types';
+import { extractTypeFromUrn } from '../../../../utils';
+
+const Container = styled.div`
+    display: flex;
+    width: 100%;
+    min-height: 500px;
+    height: 70vh;
+    border: 1px solid #f0f0f0;
+    border-radius: 4px;
+`;
+
+const SearchSection = styled.div`
+    flex: 2;
+    height: 100%;
+    overflow: hidden;
+    display: flex;
+    flex-direction: column;
+`;
+
+const SubSearchSection = styled.div`
+    flex: 1;
+    height: 100%;
+    overflow: hidden;
+    display: flex;
+    flex-direction: column;
+`;
+
+const CurrentSection = styled.div`
+    flex: 1;
+    width: 40%;
+    border-left: 1px solid #f0f0f0;
+    display: flex;
+    flex-direction: column;
+`;
+
+const SectionHeader = styled.div`
+    padding-left: 20px;
+    margin-top: 10px;
+    font-size: 16px;
+    font-weight: 500;
+    color: #666;
+`;
+
+const ScrollableContent = styled.div`
+    flex: 1;
+    overflow: auto;
+    padding: 20px;
+`;
+
+const SelectedItem = styled.div`
+    display: flex;
+    padding: 8px;
+    border-radius: 4px;
+    margin-bottom: 8px;
+    align-items: center;
+    border: 1px solid #f0f0f0;
+    justify-content: space-between;
+
+    &:hover {
+        background-color: #fafafa;
+    }
+`;
+
+const IconWrapper = styled.div`
+    cursor: pointer;
+`;
+
+const INITIAL_SELECTED_ENTITIES = [] as EntityAndType[];
+
+// New interface for the generic component
+interface SearchSelectUrnInputProps {
+    allowedEntityTypes: EntityType[];
+    isMultiple: boolean;
+    selectedValues: string[];
+    updateSelectedValues: (values: string[] | number[]) => void;
+}
+
+// Generic component that doesn't depend on StructuredProperty
+export function SearchSelectUrnInput({
```

Comment:
since we don't use this in prompts how do you feel about moving these components out to `src/app/entity/shared/components/styled/StructuredProperty`?

Reply:
or this one can be in an even more generic place since it doesn't have to depend on structured props?

Reply:
moved!